### ISSUE TITLE
Watch for Tesla app does not work

### ISSUE THREAD

**Issue by iCar-Stefan** [Kemmey/Kemmey-TeslaWatch-Public#81]
Hello,

when I start the app on my watch it says my battery is good for 403 km, and in my car, it says 328 km. Not quite the same. If I try to open tune or frun they do not work. Pretty much nothing works. What I could do after a few attempts was to Open charge port, but that was it.

Stefan

**Comment by Kemmey** [comment]
Range is duplicate of https://github.com/Kemmey/Kemmey-TeslaWatch-Public/issues/40

For the trunk and frunk actions, please verify both watch and car have valid internet connectivity. The app only works via internet connectivity, not Bluetooth.

**Comment by iCar-Stefan** [comment]
Hello,

If I understand you correctly, I should change the Range showed in my Tesla Model S?

The watch has its own e-sim and the car has the internet LTE, so Yes, they both have Internet connectivity. What else could be wrong?

Med vänlig hälsning / Best regards
Stefan Nordin


Från: Kim Hansen <notifications@github.com>
Svara till: Kemmey/Kemmey-TeslaWatch-Public <reply@reply.github.com>
Datum: tisdag 4 augusti 2020 18:57
Till: Kemmey/Kemmey-TeslaWatch-Public <Kemmey-TeslaWatch-Public@noreply.github.com>
Kopia: Stefan Nordin <stefan.nordin@auxilium.se>, Author <author@noreply.github.com>
Ämne: Re: [Kemmey/Kemmey-TeslaWatch-Public] Watch for Tesla app does not work (#81)


Range is duplicate of #40<https://github.com/Kemmey/Kemmey-TeslaWatch-Public/issues/40>

For the trunk and frunk actions, please verify both watch and car have valid internet connectivity. The app only works via internet connectivity, not Bluetooth.

**Comment by Kemmey** [comment]
The range will show correctly with the next release of the app. You can get it now by joining the beta program here: https://testflight.apple.com/join/FdzFOLZv

Can you successfully open trunks etc if you disable Bluetooth on your phone? 

The apps title-bar will in the right corner show the connectivity status and age of data. What does it show? When everything is working properly it should  alternate between online and loading and 0s up to 6s. The number is the age of the data - should only be a few seconds, but can be very old if car is asleep or not connected.

**Comment by iCar-Stefan** [comment]
When is the next release available through Appstore?

I will try tomorrow when I get to my car. But, that is no solution, I can not disable Bluetooth since I use it for many other things, like AirPods, between the car and the phone and other things.

**Comment by Kemmey** [comment]
Absolutely - disabling Bluetooth is only for testing. It was not meant to be a solution. But I'm very interested in the reusing - if you can do all actions with the Tesla app on your phone with Bluetooth disabled - but still can't with the watch. That would really help me narrow down the issue that is plaguing you. 

V. 1.0.6 is mostly feature complete - I still need to get all localizations in place, and have one or two feature requests I'd like to include as quick fixes. I can't give a release date as it is fully dependent on apples approval. But this week I'm camping, so not until end of next week at the earliest.

**Comment by iCar-Stefan** [comment]
All right, have been testing...

No difference with Bluetooth on or off. The following functions work:
     - Ventilation on/off
     - Lock/unlock car
     - Open charge port
     - Show where you are on map
     - Blink lights
     - Use horn to honk
What does not work:
     - Percentage of battery power (will be fixed in next version)
     - Open trunk
     - Open frunk
     - Open windows (ventilate)

Next, I tried with my iPhone, and guess what? From the phone, I can NOT open trunk/frunk/windows... I can open the trunk and frunk from the key fob.

Is there a setting somewhere that actually refrain the phone (and probably watch) from doing this? Or is it the car that does have a software issue?

I have a Tesla Model S 85, 2015 model, with LTE internet connection. I have an iPhone Xs Max and a AppleWatch series 5 with its own e-sim.

Do you have any ideas?

**Comment by Kemmey** [comment]
Thank you so much for testing!

Ok, the windows are expected. I've heard from multiple sources now that no model s vehicles have window control. For vehicles with sunroof, next version will allow venting and closing that. I'll remove the window control entirely for model s vehicles. 

The trunks are new - I've not had any reports of those not working. You can disable mobile access from the car entirely, but I don't think you can disable only the trunks. 

I DO have one other report of the frunk not working - I'm yet to receive feedback on which model that was. Might be related. 

I do believe I get feedback from the api if trunks can be actuated - I don't react to those values currently though, as I believed all vehicles could operate the trunks. Makes no sense otherwise. But I only have a model 3 - so rely on users to give feedback from the other vehicles. 

I'll update the store listing with next update to reflect these findings. 

I'm really sorry you were misled - but the app can only do what the official app can do without Bluetooth. Please accept my apologies and request a refund with Apple - I unfortunately can't give you that, but they are always forthcoming with refunds.

**Comment by iCar-Stefan** [comment]
Thanks but I do not need a refund.

Did you have any idea about it being a problem with my specific car, since it does not work from my iPhone either? I guess it should, should it not?

**Comment by Kemmey** [comment]
Only from what I can google. I found this old post: https://forums.tesla.com/discussion/161720/app-won-t-open-doors-trunk-frunk-version-8-3

**Comment by Kemmey** [comment]
Also this:


https://teslamotorsclub.com/tmc/threads/app-wont-unlock-doors.108684/#fc-post-2879605

You don't seem to be alone with this problem. 

Have you had any repairs done? Tesla seems to suggest a reboot of the car computer and re-install if the app - neither of which seemed to help the people affected. 

I would contact your local service center to have them debug the iPhone app - then the watch app should "just work"...

**Comment by iCar-Stefan** [comment]
Thanks, I'll try that!

**Comment by iCar-Stefan** [comment]
I got Tesla support to help me OTA, since they replaced my MCU a while back, and it works perfectly in both the app AND on the Watch. 

Thanks so much for helping me, and I am very sorry I blamed your app, when the fault was totally on Tesla!

**Comment by Kemmey** [comment]
No worries - happy it got sorted! :-D 

But still no window control, right?

**Comment by iCar-Stefan** [comment]
Right

Skickat från iPhone

Med vänlig hälsning
Stefan Nordin
________________________________
Från: Kim Hansen <notifications@github.com>
Skickat: Wednesday, August 5, 2020 5:03:53 PM
Till: Kemmey/Kemmey-TeslaWatch-Public <Kemmey-TeslaWatch-Public@noreply.github.com>
Kopia: Stefan Nordin <stefan.nordin@auxilium.se>; Author <author@noreply.github.com>
Ämne: Re: [Kemmey/Kemmey-TeslaWatch-Public] Watch for Tesla app does not work (#81)


No worries - happy it got sorted! :-D

But still no window control, right?

**Comment by remimarents** [comment]
I am able to open trunk, but not close it. Model S, TestFlight beta app.

**Comment by Kemmey** [comment]
@remimarents Have you tried pushing the open trunk button again? It will still say open trunk - but it's the same command for close - so it really should close. I'll have to update the logic to heuristically guess if it's opening or closing. Command is actually "actuate" trunk - so all I do is send that command :-D

**Comment by remimarents** [comment]
It worked as you said. Thanks.

> @remimarents Have you tried pushing the open trunk button again? It will still say open trunk - but it's the same command for close - so it really should close. I'll have to update the logic to heuristically guess if it's opening or closing. Command is actually "actuate" trunk - so all I do is send that command :-D

**Comment by rinsvelt** [comment]
> The range will show correctly with the next release of the app. You can get it now by joining the beta program here: https://testflight.apple.com/join/FdzFOLZv
> 

Hi, I bought & installed the app yesterday and the range is still not showing correctly. When is the release expected that fixes this?
